services/src/datasets/listing.rs

```diff
@@ -21,12 +21,14 @@ use validator::{Validate, ValidationError};
 #[derive(Debug, Serialize, Deserialize, Clone, PartialEq, ToSchema)]
 #[serde(rename_all = "camelCase")]
 pub struct DatasetListing {
+    #[schema(value_type = crate::api::model::datatypes::DatasetId)]
     pub id: DatasetId,
     pub name: DatasetName,
     pub display_name: String,
     pub description: String,
     pub tags: Vec<String>,
     pub source_operator: String,
+    #[schema(value_type = crate::api::model::operators::TypedResultDescriptor)]
```

Comment:
maybe move the DatasetListing to api/model or create a separate type? isn't it dangerous to assume the api typedresultdescritpr and the other typedresultdescriptor have the same schema?

Reply:
Maybe but then we also are not able to use custom schema impls instead of derives.
We will notice it if the integration tests from the libs won't work.
Some types are not optimal yet, but it is cumbersome to make everything very good.